src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -632,24 +642,51 @@ void crop_in_place_optimization::update_in_place_crop_padding_along_feature(cons
                                                                                  const tensor offsets,
                                                                                  size_t crop_axis,
                                                                                  bool is_runtime) {
-    auto crop_axis_legacy = crop_axis;
-    if (crop_axis_legacy >= 2) {
-        auto spatial_axis = crop_axis_legacy - 2;
-        // Default and minimum number of dimensions is 4
-        auto spatial_size = std::max<size_t>(crop_layout.get_partial_shape().size(), 4) - 2;
-        crop_axis_legacy = spatial_size - spatial_axis - 1 + 2;
-    }
+    auto convert_axis_to_legacy = [](size_t axis, size_t rank) {
+        auto axis_legacy = axis;
+        if (axis_legacy >= 2) {
+            auto spatial_axis = axis_legacy - 2;
+            // Default and minimum number of dimensions is 4
+            auto spatial_size = std::max<size_t>(rank, 4) - 2;
+            axis_legacy = spatial_size - spatial_axis - 1 + 2;
+        }
+
+        return axis_legacy;
+    };
+
+    auto crop_axis_legacy = convert_axis_to_legacy(crop_axis, crop_layout.get_partial_shape().size());
+
     // If it's build-time and node is dynamic, only dynamic padding is set first
     if ((crop_layout.is_dynamic() || input_layout.is_dynamic()) && !is_runtime) {
         auto dyn_pad_sizes = tensor(0).sizes();
         dyn_pad_sizes[crop_axis_legacy] = 1;
         crop_layout.data_padding.set_dynamic_pad(tensor(dyn_pad_sizes));
-        for (auto& user_layout : user_layouts) {
-            user_layout.data_padding.set_dynamic_pad(tensor(dyn_pad_sizes));
+
+        if (user_info.first && user_info.first->is_type<reshape>()) {
+            auto reshape_desc = user_info.first->as<reshape>().get_primitive();
+            auto reshape_mode = reshape_desc->mode;
+            if (reshape_mode == reshape::reshape_mode::base) {
+                user_info.second.data_padding.set_dynamic_pad(tensor(dyn_pad_sizes));
+            } else if (reshape_mode == reshape::reshape_mode::unsqueeze || reshape_mode == reshape::reshape_mode::squeeze) {
+                auto reshape_ps = user_info.second.get_partial_shape();
+                auto output_pattern = reshape_desc->output_pattern;
+
+                auto reshape_axis = crop_axis;
+                for (size_t i = 0; i < output_pattern.size(); i++) {
+                    if (output_pattern[i] <= static_cast<int64_t>(reshape_axis)) {
+                        reshape_axis += reshape_mode == reshape::reshape_mode::unsqueeze ? 1 : -1;
+                    }
+                }
+
+                auto dyn_pad_mask = tensor(0).sizes();
+                auto reshape_axis_legacy = convert_axis_to_legacy(reshape_axis, reshape_ps.size());
+                dyn_pad_mask[reshape_axis_legacy] = 1;
+                user_info.second.data_padding.set_dynamic_pad(tensor(dyn_pad_mask));
```

Comment:
@andrew-k-park  @wilson-seok  Didn't you disable squeeze/unsqueeze mode for runtime buffer fusion? But seems that with this change they can be applied too.

Reply:
Yes, we will check issued model with this change, and enable squeeze/unsqueeze mode again.

Reply:
@yeonbok @sshlyapn  I confirmed that this PR can fix TimmTwins accuracy issue without PR#25700.